java/code/src/com/redhat/rhn/frontend/action/schedule/FailedSystemsAction.java

```diff
@@ -136,6 +139,7 @@ public ActionForward handleDispatch(ListSessionSetHelper helper,
                                         ActionForm formIn, HttpServletRequest request,
                                         HttpServletResponse response) {
 
+        ensureRoleBasedAccess(new RequestContext(request).getCurrentUser(), "schedule.details", Namespace.AccessMode.W);
```

Comment:
Shouldn't this one be handled by the general web filter already? I don't get what this have in special (some more like this one on the PR, but I will not comment all)

Reply:
Ignore my comment, I got it. I was looking at the wrong place.
You put the check for `w` permission on the `handleDispatch` dispatch method, which is responsible for writing. This is one those cases where the read and write are in the same class.